### DISCUSSION TITLE
Hi, I implemented Instrumentation and using parameters.getContext() to get authentication , however , since now, getContext() is deprecated , I tried to use getGraphQLContext() instead but it returning empty object , what should I do ?

### DISCUSSION THREAD

**Discussion by bbakerman** [graphql-java/graphql-java#2815] [General]
Hi, I implemented Instrumentation and using parameters.getContext() to get authentication , however , since now, getContext() is deprecated , I tried to use getGraphQLContext() instead but it returning empty object , what should I do ?
thx

@Slf4j
@RequiredArgsConstructor
public class RequestLoggingInstrumentation extends SimpleInstrumentation {

    @Override
    public InstrumentationContext<ExecutionResult> beginExecution(
            InstrumentationExecutionParameters parameters) {
        GraphQLServletContext context = parameters.getContext();
        HttpServletRequest request = context.getHttpServletRequest();
        KeycloakSecurityContext keycloakSecurityContext = (KeycloakSecurityContext) 
        request.getAttribute(KeycloakSecurityContext.class.getName());

_Originally posted by @David-Java-stack in https://github.com/graphql-java/graphql-java/issues/1387#issuecomment-1112595448_

**Comment by bbakerman** [comment]
> I tried to use getGraphQLContext() instead but it returning empty object 

`graphql.GraphQLContext` is a map thing that needs to be setup in your ExecutionInput - if you have some special object like `GraphQLServletContext` then you need to put it into this map at query setup time.

As a guess I think `GraphQLServletContext` is coming from some framework built on top of graphql-java.  Is that true?

If so you might need to speak / understand that framework better

**Reply by bbakerman** [reply]
A google says https://javadoc.io/doc/com.graphql-java-kickstart/graphql-java-servlet/9.0.0/graphql/servlet/context/GraphQLServletContext.html

You might need to speak to the kickstart people since perhaps they have not moved to the new context object.  If they dont, you will need to call the deprecated method

**Comment by David-Java-stack** [comment]
thx for your quick reply,
I opened the below there 
https://github.com/graphql-java-kickstart/graphql-java-servlet/discussions/438
